agent/envoyextensions/builtin/otel-access-logging/otel-access-logging.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package otelaccesslogging
+
+import (
+	"fmt"
+
+	envoy_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/hashicorp/consul/api"
+	ext_cmn "github.com/hashicorp/consul/envoyextensions/extensioncommon"
+	"github.com/hashicorp/go-multierror"
+)
+
+type otelAccessLogging struct {
+	ext_cmn.BasicExtensionAdapter
+
+	// ProxyType identifies the type of Envoy proxy that this extension applies to.
+	// The extension will only be configured for proxies that match this type and
+	// will be ignored for all other proxy types.
+	ProxyType api.ServiceKind
+	// InsertOptions controls how the extension inserts the filter.
+	InsertOptions ext_cmn.InsertOptions
+	// Config holds the extension configuration.
+	Config AccessLog
+}
+
+var _ ext_cmn.BasicExtension = (*otelAccessLogging)(nil)
+
+func Constructor(ext api.EnvoyExtension) (ext_cmn.EnvoyExtender, error) {
+	auth, err := newOTELAccessLogging(ext)
+	if err != nil {
+		return nil, err
+	}
+	return &ext_cmn.BasicEnvoyExtender{
+		Extension: auth,
+	}, nil
+}
+
+// CanApply indicates if the ext-authz extension can be applied to the given extension runtime configuration.
+func (a *otelAccessLogging) CanApply(config *ext_cmn.RuntimeConfig) bool {
+	return config.Kind == api.ServiceKindConnectProxy
+}
+
+// PatchClusters modifies the cluster resources for the ext-authz extension.
+//
+// If the extension is configured to target an ext-authz service running on the local host network
+// this func will insert a cluster for calling that service. It does nothing if the extension is
+// configured to target an upstream service because the existing cluster for the upstream will be
+// used directly by the filter.
+func (a *otelAccessLogging) PatchClusters(cfg *ext_cmn.RuntimeConfig, c ext_cmn.ClusterMap) (ext_cmn.ClusterMap, error) {
+	cluster, err := a.Config.CommonConfig.toEnvoyCluster(cfg)
+	if err != nil {
+		return c, err
+	}
+	if cluster != nil {
+		c[cluster.Name] = cluster
+	}
+	return c, nil
+}
+
+// PatchFilters inserts an ext-authz filter into the list of network filters or the filter chain of the HTTP connection manager.
+func (a *otelAccessLogging) PatchFilters(cfg *ext_cmn.RuntimeConfig, filters []*envoy_listener_v3.Filter, isInboundListener bool) ([]*envoy_listener_v3.Filter, error) {
+	// The ext_authz extension only patches filters for inbound listeners.
+	if !isInboundListener {
+		return filters, nil
+	}
+
+	a.configureInsertOptions()
+
+	extAuthzFilter, err := a.Config.toEnvoyNetworkFilter(cfg)
+	if err != nil {
+		return filters, err
+	}
+	return ext_cmn.InsertNetworkFilter(filters, extAuthzFilter, a.InsertOptions)
```

Comment:
My understanding from the [Aperture integration docs](https://docs.fluxninja.com/integrations/istio/) is that we need to patch the existing `http_connection_manager` filter to add an `access_log` entry. So at a high level I think what this func should actually be doing is:
1. it should be implementing `PatchFilter` instead of `PatchFilters` because we are only interested in patching  the HCM.
2. it should get a handle to the existing HCM
3. generate an Envoy `access_log` object from its config
4. add the `access_log` object to the HCM
5. return a new copy of the updated HCM

The snippet below is incomplete but something to the effect of:

```go
func (e *otelAccessLogging) PatchFilter(p ext_cmn.FilterPayload) (*envoy_listener_v3.Filter, bool, error) {
	httpConnMgr, err := ext_cmn.GetHTTPConnection(p.Message...)
	accessLog, err := e.toEnvoyAccessLog(p.RuntimeConfig)
	httpConnMgr.AccessLog = append(httpConnMgr.AccessLog, accessLog)
	newHCM, err := ext_cmn.MakeFilter("envoy.filters.network.http_connection_manager", httpConnMgr)
	return newHCM, true, nil
}
```

Reply:
Made the changes as suggested. Thanks